cg/apps/demultiplex/sample_sheet/api.py

```diff
@@ -90,8 +95,15 @@ def validate_sample_sheet(
             message: str = f"Sample sheet with path {sample_sheet_path} does not exist"
             LOG.error(message)
             raise SampleSheetError(message)
-        self.validator.validate_sample_sheet_from_file(
-            file_path=sample_sheet_path, bcl_converter=bcl_converter
+        sample_sheet_content: list[list[str]] = ReadFile.get_content_from_file(
+            file_format=FileFormat.CSV, file_path=sample_sheet_path
+        )
+        if not bcl_converter:
+            bcl_converter: str = FLOW_CELL_SAMPLE_TO_BCL_CONVERTER[
+                get_sample_type_from_content(sample_sheet_content)
+            ]
+        self.validator.validate_sample_sheet_from_content(
```

Comment:
Design: ideally, the sample sheet would be directly parsed into a pydantic model containing the validation. I think we discussed this before and decided to add it to the backlog for a future refactoring

Reply:
marking this as out of scope and resolving